uaclient/entitlements/base.py

```diff
@@ -273,8 +310,36 @@ def can_enable(self) -> "Tuple[bool, Optional[CanEnableFailure]]":
                     ),
                 )
 
+        if self.required_services:
+            if not self.check_required_services_active():
+                return (
+                    False,
+                    CanEnableFailure(
+                        CanEnableFailureReason.INACTIVE_REQUIRED_SERVICES
+                    ),
+                )
+
         return (True, None)
 
+    def check_required_services_active(self):
+        """
+        Check if all required services are active
+
+        :return:
+            True if all required services are active
+            False is at least one of the required services is disabled
+        """
+        from uaclient.entitlements import ENTITLEMENT_CLASS_BY_NAME
+
+        for required_service in self.required_services:
+            ent_cls = ENTITLEMENT_CLASS_BY_NAME.get(required_service)
+            if ent_cls:
```

Comment:
The logic here is a silent failure if we've typo'd required_services in the static class definition. It it worth us not being defensive here so we raise a noisy error or maybe if we log a warning if required_service not in ENTITLEMENT_CLASS_BY_NAME? Same question for dependent_services logic.

Reply:
Yes, good point. I think we should fail loudly here